rq/job.py

```diff
@@ -498,8 +499,7 @@ def restore(self, raw_data):
         self.failure_ttl = int(obj.get('failure_ttl')) if obj.get('failure_ttl') else None  # noqa
         self._status = obj.get('status').decode() if obj.get('status') else None
 
-        dependency_id = obj.get('dependency_id', None)
-        self._dependency_ids = [as_text(dependency_id)] if dependency_id else []
+        self._dependency_ids = json.loads(as_text(obj.get('dependency_ids', '[]')))
```

Comment:
For backwards compatibility purposes, this should also check for `dependency_id`.